velox/dwio/dwrf/reader/ReaderBase.cpp

```diff
@@ -135,7 +112,7 @@ ReaderBase::ReaderBase(
       fileLength_,
       "Corrupted file, Post script size is invalid");
 
-  if (fileFormat == FileFormat::DWRF) {
+  if (options.fileFormat() == FileFormat::DWRF) {
```

Comment:
This is breaking some internal use that does not call `setFileFormat`.  When `options.fileFormat() != FileFormat::ORC`, we should default to `DWRF` for `DwrfReader`.  Let us do it in this class whenever we access `fileFormat()` (you can use a helper method to do this), and we can remove the extra `setFileFormat` calls added.

Reply:
I plan to add the following method in `ReaderBase.h`. Can this solve the problem?
```
dwio::common::FileFormat fileFormat() const {
    if (options_.fileFormat() == dwio::common::FileFormat::ORC) {
      return dwio::common::FileFormat::ORC;
    }

    return dwio::common::FileFormat::DWRF;
}
```

Reply:
Hi @Yuhta I have modified the corresponding code according to the commit comments. and I have rebased the code. The diff file can be found [here](https://github.com/facebookincubator/velox/commit/82ab86e65411cd2b57e251aef81f0efedd6b253e)